onedal/linear_model/linear_model.py

```diff
@@ -193,12 +161,66 @@ def __init__(
         super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
 
     def fit(self, X, y, queue=None):
-        return super()._fit(
-            X, y, self._get_backend("linear_model", "regression", None), queue
-        )
+        """
+        Fit linear model.
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training data.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values. Will be cast to X's dtype if necessary.
+
+        queue : dpctl.SyclQueue
+            If not None, use this queue for computations.
+
+        Returns
+        -------
+        self : object
+            Fitted Estimator.
+        """
+        module = self._get_backend("linear_model", "regression")
+
+        # TODO Fix _check_X_y to make sure this conversion is there
```

Comment:
not the scope of this PR. it is better to have tickets instead of TODOs

Reply:
ticket is created